kornia/filters/kernels.py

```diff
@@ -15,18 +15,31 @@ def normalize_kernel2d(input: Tensor) -> Tensor:
     return input / (norm.unsqueeze(-1).unsqueeze(-1))
 
 
-def gaussian(window_size: int, sigma: float) -> Tensor:
-    device, dtype = None, None
+def gaussian(window_size: int, sigma: Union[Tensor, float]) -> Tensor:
     if isinstance(sigma, Tensor):
-        device, dtype = sigma.device, sigma.dtype
+        device = sigma.device
+        dtype = sigma.dtype
+        if sigma.shape == torch.Size(()) or len(sigma) == 1:
```

Comment:
have a better way to check if a tensor have a unique value?

Reply:
i think len == 0